alfresco-health-processor-platform/src/test/java/eu/xenit/alfresco/healthprocessor/fixer/solr/SolrMissingNodeFixerPluginTest.java

```diff
@@ -185,4 +190,129 @@ void fix_missing_nodes_throws_reindex() throws IOException {
+
+        when(executor.executeAsyncNodeCommand(eq(endpoint1), any(), eq(SolrNodeCommand.REINDEX_TRANSACTION)))
+                .thenReturn(new SolrActionResponse(true ,"scheduled"));
+
+        Set<NodeFixReport> nodeFixReports = missingNodeFixerPlugin.fix(SolrIndexValidationHealthProcessorPlugin.class,
+                set(healthReport1, healthReport2, healthReport3));
+
+        // Only one REINDEX_TRANSACTION should have been sent for healthreport1 and healthreport2
+        verify(executor, Mockito.atMost(2)).executeAsyncNodeCommand(eq(endpoint1), any(NodeRef.Status.class), eq(SolrNodeCommand.REINDEX_TRANSACTION));
+        verifyNoMoreInteractions(executor);
+
+        assertEquals(Collections.singletonList(NodeFixStatus.SUCCEEDED),
+                nodeFixReports.stream().filter(fixReport -> fixReport.getHealthReport().equals(healthReport1))
+                        .map(NodeFixReport::getFixStatus).collect(
+                                Collectors.toList()));
+        assertEquals(Collections.singletonList(NodeFixStatus.SUCCEEDED),
+                nodeFixReports.stream().filter(fixReport -> fixReport.getHealthReport().equals(healthReport2))
+                        .map(NodeFixReport::getFixStatus).collect(
+                                Collectors.toList()));
+    }
+
+    @Test
+    void reindex_retried_after_fail() throws IOException{
+        NodeHealthReport healthReport1 = TestReports.unhealthy();
+        NodeHealthReport healthReport2 = TestReports.unhealthy();
+        SearchEndpoint endpoint1 = new SearchEndpoint(URI.create("http://empty/solr/core1/"));
+        NodeRef.Status nodeRefStatus1 = new Status(123L, healthReport1.getNodeRef(), "1", 1L, false);
+        NodeRef.Status nodeRefStatus2 = new Status(1234L, healthReport2.getNodeRef(), "1", 1L, false);
+        NodeIndexHealthReport nodeIndexHealthReport1 = new NodeIndexHealthReport(
+                IndexHealthStatus.NOT_FOUND,
+                nodeRefStatus1,
+                endpoint1
+        );
+        NodeIndexHealthReport nodeIndexHealthReport2 = new NodeIndexHealthReport(
+                IndexHealthStatus.NOT_FOUND,
+                nodeRefStatus2,
+                endpoint1
+        );
+        healthReport1.data(NodeIndexHealthReport.class).add(nodeIndexHealthReport1);
+        healthReport2.data(NodeIndexHealthReport.class).add(nodeIndexHealthReport2);
+
+        when(executor.executeAsyncNodeCommand(eq(endpoint1), any(), eq(SolrNodeCommand.REINDEX_TRANSACTION)))
+                .thenReturn(new SolrActionResponse(false ,"failure"));
+
+        Set<NodeFixReport> nodeFixReports = missingNodeFixerPlugin.fix(SolrIndexValidationHealthProcessorPlugin.class,
+                set(healthReport1, healthReport2));
+
+        // Only one REINDEX_TRANSACTION should have been sent
+        verify(executor, Mockito.atLeast(2)).executeAsyncNodeCommand(eq(endpoint1), any(NodeRef.Status.class), eq(SolrNodeCommand.REINDEX_TRANSACTION));
```

Comment:
If only one should have been sent, then why do you assert that this happens **at least** 2 times?